lib/chef/provider/aws_security_group.rb

```diff
@@ -284,11 +317,12 @@ def update_outbound_rules(sg, vpc)
   def update_rules(desired_rules, actual_rules_list, authorize: nil, revoke: nil)
     actual_rules = {}
     actual_rules_list.each do |rule|
+      rule = Hash[rule.each_pair.to_a]
```

Comment:
What is the expected type of `rule` here?  If it responds to #each_pair I'm a bit surprised that we need to us this to coerce it to a Hash at all, or is there something going on here I'm not seeing?